snuba/utils/streams/consumers/consumer.py

```diff
@@ -99,7 +108,7 @@ def commit(self) -> Mapping[TStream, TOffset]:
 
         Raises a ``RuntimeError`` if called on a closed consumer.
         """
-        return self.__backend.commit()
+        return self.__commit_retry_policy.call(self.__backend.commit)
```

Comment:
I'm struggling to come up with an approach to testing this that doesn't rely on implementing a `ConsumerBackend` subclass and either filling out most of the method bodies with `NotImplementedError` or actually implementing a test backend (which we should do, but will be a time commitment to get the implementation right.)

Reply:
Use mocks for the backend ? Anyway it is not worse than having a fake ConsumerBackend and you can instruct it to throw when you want and you can assert how many times a method is called.
https://docs.python.org/3/library/unittest.mock.html

Unless of course you want to preserve the functionality of the ConsumerBackend

Reply:
I guess this is the best option for now — I just don't like how this converts the backend to `Any` and all of the sudden the mocked behavior in tests could violate the backend interface contract without warning. 😕